server/src/db/filesystem_db.rs

```diff
@@ -27,6 +27,37 @@ impl FileSystemDb {
         }
     }
 
+    fn provided_entity<T: DbEntity>(&self, user_id: &Id<User>, provided_id: &Id<T>) -> bool {
+        let provided_entity_path = self.get_path(user_id, Some(provided_id));
+        match std::fs::metadata(provided_entity_path).map_err(|e| e.kind()) {
+            Ok(_) => true,
+            Err(std::io::ErrorKind::NotFound) => {
+                println!("Provided entity not found!");
+                false
+            }
+            _ => false,
```

Comment:
This function, answering a yes-or-no question, shouldn't really do error handling. The caller is responsible to decide whether `true`, `false` or neither are considered errors and what to do in such a case.